taxi/parser/parser.go

```diff
@@ -0,0 +1,79 @@
+package parser
+
+import (
+	"strconv"
+	"strings"
+	"taxi/record"
+)
+
+// Parse can parse the taxi logs. We usually used it  
+// when caclurating some information from taxi logs.
+func Parse(logs string) ([]record.Record, error) {
+	var records []record.Record
+	for _, log := range strings.Split(logs, "<LF>") {
+		// Skip a blank line
+		if log == ""{
+			break
+		}
+		convertedLog := strings.Split(log, " ")
+		var rec record.Record
+		time, err := createTimeObject(convertedLog[0])
+		if err != nil {
+			return []record.Record{}, err
+		}
+		rec.Time = time
+		distance, err := strconv.ParseFloat(convertedLog[1], 64)
+		if err != nil {
+			return []record.Record{}, err
+		}
+		rec.Distance = distance
+		records = append(records, rec)
+	}
+	return records, nil
+}
+
+// createTimeObject converts the timestamp into Time class.
+// time.Hours is hours and time.Minutes is minutes.
+func createTimeObject(timestamp string) (record.Time, error) {
```

Comment:
Confusing name. Though this creates a record.Time object, a string variable is also called timeObject in this function.

Reply:
createTimeObject -> newTime